modules/social_features/social_topic/src/Plugin/GraphQL/SchemaExtension/TopicSchemaExtension.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Drupal\social_topic\Plugin\GraphQL\SchemaExtension;
+
+use Drupal\graphql\GraphQL\ResolverBuilder;
+use Drupal\graphql\GraphQL\ResolverRegistryInterface;
+use Drupal\graphql\Plugin\GraphQL\SchemaExtension\SdlSchemaExtensionPluginBase;
+
+/**
+ * Adds topic data to the Open Social GraphQL API.
+ *
+ * @SchemaExtension(
+ *   id = "social_topic_schema_extension",
+ *   name = "Open Social - Topic Schema Extension",
+ *   description = "GraphQL schema extension for Open Social topic data.",
+ *   schema = "open_social"
+ * )
+ */
+class TopicSchemaExtension extends SdlSchemaExtensionPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function registerResolvers(ResolverRegistryInterface $registry) {
+    $builder = new ResolverBuilder();
+
+    $this->addQueryFields($registry, $builder);
+    $this->addTopicFields($registry, $builder);
+  }
+
+  /**
+   * Registers type and field resolvers in the shared registry.
+   *
+   * @param \Drupal\graphql\GraphQL\ResolverRegistryInterface $registry
+   *   The resolver registry.
+   * @param \Drupal\graphql\GraphQL\ResolverBuilder $builder
+   *   The resolver builder.
+   */
+  protected function addTopicFields(ResolverRegistryInterface $registry, ResolverBuilder $builder) {
+    $registry->addFieldResolver('Topic', 'id',
+      $builder->produce('entity_uuid')
+        ->map('entity', $builder->fromParent())
+    );
+
+    $registry->addFieldResolver('Topic', 'title',
+      $builder->compose(
+        $builder->produce('entity_label')
+          ->map('entity', $builder->fromParent()),
+        $builder->produce('uppercase')
+          ->map('string', $builder->fromParent())
+      )
+    );
+
+    $registry->addFieldResolver('Topic', 'author',
+      $builder->produce('entity_owner')
+        ->map('entity', $builder->fromParent())
+    );
+
+    $registry->addFieldResolver('Topic', 'bodyHtml',
+      $builder->fromPath('entity:node', 'body.0.value')
```

Comment:
This needs rendering. See https://github.com/goalgorilla/open_social/blob/55e5b28c2750bd9d4a5da1b18c4c4836fed67a65/modules/social_features/social_comment/src/Plugin/GraphQL/SchemaExtension/CommentSchemaExtension.php#L50

The field value may contain Drupal specific tags that are not valid HTML.

Reply:
Yeah, but it will return a lot of HTML, that is probably not needed,

E.g. with how it is implemented in comments:
![Снимок экрана 2021-07-06 в 17 14 27](https://user-images.githubusercontent.com/50984627/124615129-a1194c00-de7d-11eb-86d6-ffcc1bc8b6dc.png)

E.g. with `$builder->fromPath('entity:node', 'body.0.value')`:
![Снимок экрана 2021-07-06 в 17 15 25](https://user-images.githubusercontent.com/50984627/124615260-c443fb80-de7d-11eb-9b3d-34131899f830.png)

But if you think the first example is correct then I'll change it.

Reply:
The `path` version will break as soon as you add an image though. So in my opinion that's not an option.

If the excess HTML wrapping is an issue when rendering the field, then we should figure out a way to properly configure fields for rendering in GraphQL (e.g. without label and without extra field wrappers). That probably means we'll need to add extra view modes for all content and fields though, so I think that's outside of the scope for this PR and an improvement we can make in the future.

The extra included div is slightly unfortunate in that it uses quite generic classes but that's something that people should be able to work around relatively easily for the time being.

Reply:
Implemented as in `comments` for now.